cupy_backends/cuda/libs/cusolver.pyx

```diff
@@ -1508,6 +1678,359 @@ cpdef zsytrf(intptr_t handle, int uplo, int n, size_t A, int lda,
+                               size_t dX, int lddx, size_t dwork) except? -1:
+    cdef size_t lwork
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnDXgels_bufferSize(
+            <Handle>handle, m, n, nrhs, <double*>dA, ldda,
+            <double*>dB, lddb, <double*>dX, lddx, <void*>dwork, &lwork)
+    check_status(status)
+    return lwork
+
+cpdef size_t dhgels_bufferSize(intptr_t handle, int m, int n, int nrhs,
+                               size_t dA, int ldda, size_t dB, int lddb,
+                               size_t dX, int lddx, size_t dwork) except? -1:
+    cdef size_t lwork
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnDHgels_bufferSize(
+            <Handle>handle, m, n, nrhs, <double*>dA, ldda,
+            <double*>dB, lddb, <double*>dX, lddx, <void*>dwork, &lwork)
+    check_status(status)
+    return lwork
+
+cpdef size_t ssgels_bufferSize(intptr_t handle, int m, int n, int nrhs,
+                               size_t dA, int ldda, size_t dB, int lddb,
+                               size_t dX, int lddx, size_t dwork) except? -1:
+    cdef size_t lwork
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnSSgels_bufferSize(
+            <Handle>handle, m, n, nrhs, <float*>dA, ldda,
+            <float*>dB, lddb, <float*>dX, lddx, <void*>dwork, &lwork)
+    check_status(status)
+    return lwork
+
+cpdef size_t sxgels_bufferSize(intptr_t handle, int m, int n, int nrhs,
+                               size_t dA, int ldda, size_t dB, int lddb,
+                               size_t dX, int lddx, size_t dwork) except? -1:
+    cdef size_t lwork
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnSXgels_bufferSize(
+            <Handle>handle, m, n, nrhs, <float*>dA, ldda,
+            <float*>dB, lddb, <float*>dX, lddx, <void*>dwork, &lwork)
+    check_status(status)
+    return lwork
+
+cpdef size_t shgels_bufferSize(intptr_t handle, int m, int n, int nrhs,
+                               size_t dA, int ldda, size_t dB, int lddb,
+                               size_t dX, int lddx, size_t dwork) except? -1:
+    cdef size_t lwork
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnSHgels_bufferSize(
+            <Handle>handle, m, n, nrhs, <float*>dA, ldda,
+            <float*>dB, lddb, <float*>dX, lddx, <void*>dwork, &lwork)
+    check_status(status)
+    return lwork
+
+cpdef int zzgels(intptr_t handle, int m, int n, int nrhs, size_t dA, int ldda,
+                 size_t dB, int lddb, size_t dX, int lddx,
+                 size_t dwork, size_t lwork, size_t dInfo):
+    cdef int iter
+    _setStream(handle)
+    with nogil:
+        status = cusolverDnZZgels(
+            <Handle>handle, m, n, nrhs, <cuDoubleComplex*>dA, ldda,
+            <cuDoubleComplex*>dB, lddb, <cuDoubleComplex*>dX, lddx,
+            <void*>dwork, lwork, &iter, <int*>dInfo)
+    check_status(status)
+    return iter
```

Comment:
OK to directly return `iter` (`niters`), the unique host scalar output, which is approved by the team.

- (+) Better usability.
- (-) Unlike `*_bufferSize` or `get*`, the interface is not canonical.

The other option was to indirectly return the output to the memory allocated outside (e.g. by `np.array`).  This option will be only for array or multiple outputs.